src/solver/solverUtils.F90

```diff
@@ -385,14 +385,14 @@ subroutine gridVelocitiesFineLevel(useOldCoor, t, sps)
        ! Set the pointers for this block.
 
        call setPointers(nn, groundLevel, sps)
-       call gridVelocitiesFineLevel_block(useOldCoor, t, sps)
+       call gridVelocitiesFineLevel_block(useOldCoor, t, sps, nn)
```

Comment:
@SichengHe `nn` is a counter used for some loops that is now passed into the `_block` subroutines, right?

Reply:
Right. We need this because we need to directly access the data block in the function.